### PR TITLE
Remove unnecessary parentheses

### PR DESCRIPTION
**Pull request by basil**
Removes grouping parentheses that are unnecessary. An earlier version of this change fixed all violations of the `UnnecessaryParentheses` Checkstyle check, but that caused new violations of Error Prone's `OperatorPrecedence` check. The current revision attempts to find a balance.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: cli/src/main/java/hudson/cli/HexDump.java
Comment (daniel-beck) [inline cli/src/main/java/hudson/cli/HexDump.java:31]: Are we confident everyone reading this knows the operator precedence rules by heart? Because this looks like something that would be more difficult to read.

Comment (basil) [reply cli/src/main/java/hudson/cli/HexDump.java:31]: Good point. This trips up Error Prone's [`OperatorPrecedence`](https://errorprone.info/bugpattern/OperatorPrecedence) check as well. I'll rework this to retain the grouping parentheses for cases like this.

**Comment by basil** [comment]
I rolled back any changes that violated Error Prone's `OperatorPrecedence` check. I think the current revision strikes a good balance.

**Review by timja — APPROVED** [review]

**Review by daniel-beck — APPROVED** [review]

**Comment by daniel-beck** [comment]
A good example that not all of these are self-evidently the correct thing to do, because there are trade-offs involved. A bit annoying that we cannot enable the auto-check though.

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
